Documentation/advanced-configuration.md

```diff
@@ -283,129 +289,90 @@ ceph osd pool set rbd pg_num 512
 
-In this example we will set the default pool `size` to three, and tell OSD
+In this example we will set the default pool `size` to two, and tell OSD
 daemons not to change the weight of OSDs on startup.
 
-Here's our custom ceph.conf:
+**WARNING**: Modify Ceph settings carefully. You are leaving the sandbox tested by Rook.
+Changing the settings could result in unhealthy daemons or even data loss if used incorrectly.
+
+### Standalone Rook
+
+Here's our custom `ceph.conf` for the OSDs:
 ```bash
 [global]
 osd crush update on start = false
 osd pool default size     = 2
 ```
 
-### Standalone Rook
-
-With our ceph.conf file created we can pass the settings to `rookd` with the
+With our ceph.conf file created we pass the settings to `rookd` with the
 `--ceph-config-override` argument, for example:
 ```bash
 rookd --ceph-config-override=ceph.conf --data-devices=sdf
 ```
 
 ### Kubernetes
-
-With rook-operator on Kubernetes we can edit the OSD DaemonSet after it has been
-created.  This will be streamlined once
-[#571](https://github.com/rook/rook/issues/571) is resolved.
-
-#### Using A ConfigMap
-
-First we will create a ConfigMap that will hold our custom ceph.conf, so it can
-be mounted as a file to be consumed in our Rook pods.
-```bash
-# Create a ConfigMap from a text file
-kubectl -n rook create configmap rook-config --from-file ceph.conf
-
-# View the resulting ConfigMap
-kubectl -n rook get configmap rook-config -o yaml
+When the Rook Operator creates a cluster, a placeholder ConfigMap is created that
+will allow you to override Ceph configuration settings. There are four categories
+of settings, one for each daemon. When the daemon pods are started, the 
+settings specified in this ConfigMap will be merged with the default settings
+generated by Rook. 
+
+The default override settings are blank. Cutting out the extraneous properties, 
+we would see the following defaults after creating a cluster:
+```
+$ kubectl -n rook get ConfigMap config-override -o yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: config-override
+  namespace: rook
+data:
+  mds: ""
+  mon: ""
+  osd: ""
+  rgw: ""
 ```
 
-If you want to make changes you can edit the ConfigMap within the Kubernetes
-cluster:
+To apply your desired configuration, you will need to update this ConfigMap.
+The next time the daemon pod(s) start, the settings will be merged with the default
+settings created by Rook.
+
 ```bash
-kubectl -n rook edit configmap rook-config
+kubectl -n rook edit configmap config-override
 ```
 
-Alternatively you can maintain a copy of the ConfigMap on your local computer
-that looks like this:
+Modify the settings and save. Each line you add should be indented from the daemon name as such:
 ```yaml
-# Filename: rook-config.yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: rook-config
+  name: config-override
+  namespace: rook
 data:
-  ceph.conf: |
+  mds: ""
+  mon: ""
+  osd: |
```

Comment:
this approach doesn't have that level of flexibility since we only have the ability to map a static name in the configmap (`osd`). We would have to generate the id of the local daemon, which gives us the translation to `[osd.0]`. To get more flexibility for subsets of osds, i think we need to wait for the change that the osd will query the api service at startup for osd-specific settings.